packages/dev/core/src/DeviceInput/webDeviceInputSystem.ts

```diff
@@ -420,12 +420,21 @@ export class WebDeviceInputSystem implements IDeviceInputSystem {
 
             const pointer = this._inputs[deviceType][deviceSlot];
             if (pointer) {
+                const deviceEvent = evt as IPointerEvent;
+                deviceEvent.inputIndex = PointerInput.Move;
+
+                if (!("movementX" in evt)) {
+                    const movementX = evt.clientX - pointer[PointerInput.Horizontal];

```

Comment:
This need to be calculated after checking that the other ones don't exist, otherwise, the computation is wasted. Ditto elsewhere.

Reply:
I would also consider evt.movementX !== undefined instead of the in for perf

Reply:
Good point, I'll make those changes